### ISSUE TITLE
`bundle gem NAME` includes files in `.circleci` regardless of the configured CI system

### ISSUE THREAD

**Issue by shonfeder** [rubygems/rubygems#7087]
### Describe the problem as clearly as you can

Running `bundle gem foo` includes files in `.circleci` regardless of the configured CI system.

This appears to be because `.circleci` is hard coded in the `gemspec` file: https://github.com/rubygems/rubygems/blob/master/bundler/lib/bundler/templates/newgem/newgem.gemspec.tt#L33C52-L33C52

### Did you try upgrading rubygems & bundler?

Yes.

### Post steps to reproduce the problem

```
$ bundle --version
Bundler version 2.4.21
$ bundle gem mwe
Creating gem 'mwe'...
MIT License enabled in config
Code of conduct enabled in config
Changelog enabled in config
Standard enabled in config
Initializing git repo in $HOME/scratch/mwe
      create  mwe/Gemfile
      create  mwe/lib/mwe.rb
      create  mwe/lib/mwe/version.rb
      create  mwe/sig/mwe.rbs
      create  mwe/mwe.gemspec
      create  mwe/Rakefile
      create  mwe/README.md
      create  mwe/bin/console
      create  mwe/bin/setup
      create  mwe/.gitignore
      create  mwe/.rspec
      create  mwe/spec/spec_helper.rb
      create  mwe/spec/mwe_spec.rb
      create  mwe/.github/workflows/main.yml # <<< because of previously configured CI choice
      create  mwe/LICENSE.txt
      create  mwe/CODE_OF_CONDUCT.md
      create  mwe/CHANGELOG.md
      create  mwe/.standard.yml
Gem 'mwe' was successfully created. For more information on making a RubyGem visit https://bundler.io/guides/creating_gem.html

$ rg '.circleci' < mwe/mwe.gemspec
        f.start_with?(*%w[bin/ test/ spec/ features/ .git .circleci appveyor Gemfile])
```

### Which command did you run?

See above.

### What were you expecting to happen?

I would not expect anything about `.circleci` to be included in the scaffold if it is not being used for CI.

In fact, I am surprised that we'd want any CI config files would be included in the gem at all, but I am also very new to Ruby.

### What actually happened?

See above.

#### Environment

```
Bundler       2.4.21
  Platforms   ruby, x86_64-linux
Ruby          3.0.6p216 (2023-03-30 revision 23a532679b406cb53c0edfc00c91c32a5ccd335a) [x86_64-linux]
  Full Path   /usr/bin/ruby
  Config Dir  /etc
RubyGems      3.3.25
  Gem Home    /usr/lib/ruby/gems/3.0.0
  Gem Path    $HOME/.gem/ruby/3.0.0:/usr/lib/ruby/gems/3.0.0
  User Home   /home/sf
  User Path   $HOME/.gem/ruby/3.0.0
  Bin Dir     /usr/bin
Tools
  Git         2.42.0
  RVM         not installed
  rbenv       rbenv 1.2.0
  chruby      not installed
```

#### Bundler Build Metadata

```
Built At          2023-10-17
Git SHA           d10b46bd15
Released Version  true
```

#### Bundler settings

```
gem.changelog
  Set for the current user ($HOME/.bundle/config): true
gem.ci
  Set for the current user ($HOME/.bundle/config): "github"
gem.coc
  Set for the current user ($HOME/.bundle/config): true
gem.linter
  Set for the current user ($HOME/.bundle/config): "standard"
gem.mit
  Set for the current user ($HOME/.bundle/config): true
gem.test
  Set for the current user ($HOME/.bundle/config): "rspec"
```

---

Thanks for maintaining the nice utility! :)

**Comment by hsbt** [comment]
@shonfeder Thanks for reporting this. I fixed this at #7101

**Comment by shonfeder** [comment]
Thanks for the quick fix! :pray:
